indico_zodbimport/modules/badges_posters.py

```diff
@@ -0,0 +1,255 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals, division
+
+import mimetypes
+from operator import attrgetter
+
+import click
+import math
+from pytz import utc
+
+from indico.core.db import db
+from indico.modules.categories import Category
+from indico.modules.events import Event
+from indico.modules.designer import event_badge_settings, category_badge_settings, DEFAULT_BADGE_SETTINGS
+from indico.modules.designer.models.templates import DesignerTemplate, TemplateType
+from indico.modules.designer.models.images import DesignerImageFile
+from indico.modules.users import User
+from indico.util.console import verbose_iterator, cformat
+from indico.util.fs import secure_filename
+from indico.util.struct.iterables import committing_iterator
+
+from indico_zodbimport import Importer, convert_to_unicode
+from indico_zodbimport.util import LocalFileImporterMixin
+
+
+ITEM_KEY_MAPPING = (('type', 'key'),
```

Comment:
probably not, but it really looks like a mapping to me so if the order doesn't matter i'd go for a dict